esrally/track/loader.py

```diff
@@ -1076,6 +1085,17 @@ def read(self, track_name, track_spec_file, mapping_dir):
 
         current_track = self.read_track(track_name, track_spec, mapping_dir, track_spec_file)
 
+        internal_user_defined_track_params = self.complete_track_params.internal_user_defined_track_params()
+        if len(internal_user_defined_track_params) > 0:
+            err_msg = "Some of your track parameter(s) {} are defined by Rally and cannot be modified.\n".format(
```

Comment:
For new code in Rally (since switching to Python 3.8 a few years back) we prefer using [f-strings](https://docs.python.org/3/tutorial/inputoutput.html#fancier-output-formatting), and take the opportunity to do so too, when dealing with older code. So this could be something along the lines of:

```python
            blacklisted_user_defined_track_params = ", ".join(opts.double_quoted_list_of(sorted(internal_user_defined_track_params)))
            err_msg = f"Some of your track parameter(s) [{blacklisted_user_defined_track_params}] are defined by Rally and cannot be modified.\n"

            ...
            raise exceptions.TrackConfigError(f"Reserved track parameters [{blacklisted_user_defined_track_params}].")
```

Reply:
Addressed in https://github.com/elastic/rally/pull/1750/commits/a3a9ff8d22d5e5be7dbb641fe589196f61a9d140.